src/NuGet.Core/NuGet.Repositories/NuGetv3LocalRepository.cs

```diff
@@ -76,6 +76,21 @@ public IEnumerable<LocalPackageInfo> FindPackagesById(string packageId)
                 });
         }
 
+        /// <summary>
+        /// Remove cached results for the given ids. This is needed
+        /// after installing a new package.
+        /// </summary>
+        public void ClearCacheForIds(IEnumerable<string> packageIds)
+        {
+            lock (_cache)
```

Comment:
This is a common pattern in the code ported from DNX, I'm not sure why they don't just use ConcurrentDictionary.